packages/epo-widget-lib/src/widgets/ColorTool/ColorTool.tsx

```diff
@@ -0,0 +1,203 @@
+    );
+
+    return setSelectedData({
+      ...selectedData,
+      filters: updatedFilters,
+    });
+  };
+
+  const handleCategorySelection = (event: FormEvent<HTMLSelectElement>) => {
+    const { value } = event.target as HTMLSelectElement;
+
+    return setSelectedData({
+      name: value,
+      filters: getDataFiltersByName(data, value),
+    });
+  };
+
+  const handleReset = () =>
+    setSelectedData({
+      ...selectedData,
+      filters: resetFilters(selectedData.filters),
+    });
+
+  const hasMultipleDatasets = data.length > 1;
+  const { t } = useTranslation();
+  const selectPlaceholder = t("colorTool.actions.select_an_object");
+  const { filters, name: selectedObjectName, caption } = selectedData;
+  const selectedCategoryName = getCategoryName(data, selectedObjectName);
+
+  return (
+    <Styled.WidgetContainer
+      ref={ref}
+      isCondensed={width < 1000}
+      hideControls={hideControls}
+    >
+      {title && <Styled.Title>{title}</Styled.Title>}
+      {selectedObjectName && (hideControls || hasMultipleDatasets) && (
+        <>
+          {hasMultipleDatasets && (
+            <Styled.Subtitle>
+              <dt>{t("colorTool.labels.object_type")}</dt>
+              <dd>{selectedCategoryName}</dd>
+              {!hideSubtitle && (
+                <>
+                  <dt>{t("colorTool.labels.selected_object")}</dt>
+                  <dd>{selectedObjectName}</dd>
+                </>
+              )}
+            </Styled.Subtitle>
+          )}
+          {!hasMultipleDatasets && !hideSubtitle && (
+            <Styled.Subtitle>
+              <dt>{t("colorTool.labels.object")}</dt>
+              <dd>{selectedObjectName}</dd>
+            </Styled.Subtitle>
+          )}
+        </>
+      )}
+      {!hideControls && (
+        <Styled.ControlsContainer>
+          {hasMultipleDatasets && (
+            <Styled.SelectionContainer>
+              <Select
+                id="astroObjectSelector"
+                placeholder={selectPlaceholder}
+                options={objectOptions}
+                onChange={handleCategorySelection}
+                value={selectedObjectName}
+                disabled={isDisabled}
+              />
+            </Styled.SelectionContainer>
+          )}
+          {filters && (
+            <>
+              <Styled.ToolsHeader id="filterLabel">
+                {t("colorTool.labels.filter")}
+              </Styled.ToolsHeader>
+              <Styled.ToolsHeader id="colorLabel">
+                {t("colorTool.labels.color")}
+              </Styled.ToolsHeader>
+              <Styled.ToolsHeader id="intensityLabel">
+                {t("colorTool.labels.color_intensity")}
+              </Styled.ToolsHeader>
+            </>
+          )}
+          {filters &&
+            filters.map((imageFilter) => {
+              const { label, isDisabled: isFilterDisabled } = imageFilter;
+
+              return (
+                <FilterControls
```

Comment:
the controls have been split out into a separate component so we don't have to match a control to it's intended filter and instead the control just fires a callback when a change is made